src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -751,7 +751,17 @@ Value ByteCodeInterpreter::interpret(ExecutionState& state, ByteCodeBlock* byteC
                 if (property.isString() && property.asString()->equals("__proto__")) {
                     willBeObject.asObject()->setPrototype(state, registerFile[code->m_loadRegisterIndex]);
                 } else {
-                    willBeObject.asObject()->defineOwnProperty(state, objPropName, ObjectPropertyDescriptor(registerFile[code->m_loadRegisterIndex], ObjectPropertyDescriptor::AllPresent));
+                    ObjectPropertyDescriptor::PresentAttribute desc = ObjectPropertyDescriptor::NotPresent;
+                    if (code->m_writable) {
+                        desc = (ObjectPropertyDescriptor::PresentAttribute)(desc | ObjectPropertyDescriptor::WritablePresent);
```

Comment:
How about emitting the ObjectPresentAttribute flag directly to the bytecodestream, therefore at this point no runtime checks would be needed to decode the flag?

Reply:
@rerobika Oh, I see, I will be fix it.

Reply:
@rerobika please, review this PR again